scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableSyntaxConfig.scala

```diff
@@ -0,0 +1,148 @@
+package scalafix.internal.config
+
+import metaconfig.{Conf, ConfError, ConfDecoder, Configured}
+import org.langmeta._
+import MetaconfigPendingUpstream.XtensionConfScalafix
+
+import scala.meta.tokens.Token
+
+case class DisableSyntaxConfig(
+    carriageReturn: Boolean = false,
+    keywords: Set[DisabledKeyword] = Set(),
+    semicolons: Boolean = false,
+    tabs: Boolean = false,
+    xml: Boolean = false
+) {
+  implicit val reader: ConfDecoder[DisableSyntaxConfig] =
+    ConfDecoder.instanceF[DisableSyntaxConfig](
+      c =>
+        (
+          c.getField(carriageReturn) |@|
+            c.getField(keywords) |@|
+            c.getField(semicolons) |@|
+            c.getField(tabs) |@|
+            c.getField(xml)
+        ).map {
+          case ((((a, b), c), d), e) =>
+            DisableSyntaxConfig(a, b, c, d, e)
+      })
+
+  def isDisabled(keyword: String): Boolean =
+    keywords.contains(DisabledKeyword(keyword))
+}
+
+object DisableSyntaxConfig {
+  val default = DisableSyntaxConfig()
+  implicit val reader: ConfDecoder[DisableSyntaxConfig] = default.reader
+}
+
+object Keyword {
+  def unapply(token: Token): Option[String] = {
+    val prefix = token.productPrefix
+    if (kwSet.contains(prefix)) {
+      Some(prefix.drop(2).toLowerCase)
+    } else {
+      None
+    }
+  }
+  val all = List(
+    "abstract",
```

Comment:
FWIW scalafmt actually has a setting to binpack argument lists consisting only of literals, it does it for numeric literals by default but for some reason I excluded strings in the default setting.